File: backend/domain-sample/src/main/java/com/plaing/domain/sample/SampleController.java
package com.plaing.domain.sample;

import com.plaing.common.response.ApiResponse;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/sample")
public class SampleController {

  private final SampleService sampleService;

  public SampleController(SampleService sampleService) {
    this.sampleService = sampleService;
  }

  @GetMapping("/health")
  public ApiResponse<String> health() {
    return ApiResponse.ok(sampleService.ping());
  }
}

File: backend/domain-sample/src/main/kotlin/com/plaing/domain/sample/SampleController.kt
package com.plaing.domain.sample

import com.plaing.common.response.ApiResponse
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/sample")
class SampleController(
    private val sampleService: SampleService,
) {
    @GetMapping("/health")
    fun health(): ApiResponse<String> = ApiResponse.ok(sampleService.ping())
}
